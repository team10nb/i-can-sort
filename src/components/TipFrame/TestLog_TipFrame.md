#### ComponentName: TipFrame

|Test No.|Date|Pass/Fail|Failed test(s)|Reason/Description|Modified version|
|--------|----|---------|--------------|------------------|----------------|
|1       | 3.07 | Fail  | All          | No code has been constructed|     |
|2       | 3.07 | Pass  | /            | /                |version 1       |

**##### Code - version 1**

```javascript
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import Backdrop from '@material-ui/core/Backdrop';

const useStyles = makeStyles((theme) => ({
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: '#fff',
    justifyContent: 'center',
  },
}));


export default function TipFrame(props) {
  const { message, size, btnColor } = props;

  const classes = useStyles();

  const styles = {
    snackbar: {
      borderRadius: 30,
      justifyContent: 'left',
      height: 220,
      width: 480,
      backgroundColor: "white",
      color: "black",
      fontSize: 18,
      paddingLeft: 40,
      paddingRight: 40,
      paddingTop: 0,

    },
    button: {
      marginRight: 220,
      marginTop: -40,
      fontWeight: 600,
      backgroundColor: btnColor,
      color: "white",
      fontSize: 15
    }
  }

  const action = (
    <Button style={styles.button}>
      OK
    </Button>
  );

  const [open, setOpen] = React.useState(true);

  const handleClose = () => {
    setOpen(false);
  };


  return (
    <div>
      <Backdrop style={size} className={classes.backdrop} open={open} onClick={handleClose}>
        <SnackbarContent
          style={styles.snackbar}
          message={message}
          action={action}
        >
        </SnackbarContent>
      </Backdrop>
    </div>
  );
}


```