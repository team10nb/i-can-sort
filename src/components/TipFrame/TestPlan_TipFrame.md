#### Component: TipFrame

|Test Name|Description|Operation|Expected Outcome|
|---------|-----------|---------|----------------|
|should contain message|||Message should be shown|
|should contain button|||Confirm button should be included|
|should match Snapshot|Compare new snapshot with old one|  | Snapshots are the same|