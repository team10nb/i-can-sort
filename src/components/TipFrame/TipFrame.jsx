/*
    Author: Yuting Jiang

    A pop up window with a backdrop, used to give tip.
*/

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import Backdrop from '@material-ui/core/Backdrop';

const useStyles = makeStyles((theme) => ({
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: '#fff',
    justifyContent: 'center',
  },
}));


export default function TipFrame(props) {
  const { message, size, btnColor } = props;

  const classes = useStyles();

  const styles = {
    snackbar: {
      borderRadius: 30,
      justifyContent: 'left',
      height: 220,
      width: 480,
      backgroundColor: "white",
      color: "black",
      fontSize: 18,
      paddingLeft: 40,
      paddingRight: 40,
      paddingTop: 0,

    },
    button: {
      marginRight: 220,
      marginTop: -40,
      fontWeight: 600,
      backgroundColor: btnColor,
      color: "white",
      fontSize: 15
    }
  }

  const action = (
    <Button style={styles.button}>
      OK
    </Button>
  );

  const [open, setOpen] = React.useState(true);

  const handleClose = () => {
    setOpen(false);
  };


  return (
    <div>
      <Backdrop style={size} className={classes.backdrop} open={open} onClick={handleClose}>
        <SnackbarContent
          style={styles.snackbar}
          message={message}
          action={action}
        >
        </SnackbarContent>
      </Backdrop>
    </div>
  );
}
