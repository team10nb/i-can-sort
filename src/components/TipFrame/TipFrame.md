### **Developer**

Yuting Jiang

Reference: MATERIAL-UI (https://material-ui.com)

###  **Introduction**

It is used to pop up a windows and show some tips.

###  **Import**

```html
import { makeStyles } from "@material-ui/core/styles";
import Button from '@material-ui/core/Button';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import Backdrop from '@material-ui/core/Backdrop';
```

###  **Props**

| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
| message| string |         | Message of the popup tip window |
| size | structure | full screen | Size of backdrop        |
| btnColor | string |  "white"   | Color of the button           |

###  **Demo**

```jsx

import React from 'react';
import Backdrop from '@material-ui/core/Backdrop';
import CircularProgress from '@material-ui/core/CircularProgress';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: '#fff',
  },
}));

const classes = useStyles();
const [open, setOpen] = React.useState(false);
const handleClose = () => {
    setOpen(false);
};
const handleToggle = () => {
    setOpen(!open);
};
const message="Hello! This is a TipFrame!";

    <div>
      <Button variant="outlined" color="primary" onClick={handleToggle}>
        Show TipFrame
      </Button>
      <Backdrop className={classes.backdrop} open={open} onClick={handleClose}>
        <TipFrame message={message} btnColor={"blue"}/>
      </Backdrop>
    </div>


```