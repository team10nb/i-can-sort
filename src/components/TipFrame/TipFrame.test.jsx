/*
    Author: Yuting Jiang
*/

import { render } from "@testing-library/react";
import TipFrame from "./TipFrame";

test("should contain message", () => {
    const message = "message";

    const { getByText } = render(<TipFrame message={message} />);
    getByText("message");
});

test("should contain button", () => {
    const message = "message";

    const { getByText } = render(<TipFrame message={message} />);
    getByText("OK");
});

test("should match Snapshot", () => {
    const message = "message";

    const content = render(<TipFrame message={message} />);
    expect(content).toMatchSnapshot();
});
