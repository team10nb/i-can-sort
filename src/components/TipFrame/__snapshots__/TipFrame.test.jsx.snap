// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-backdrop-3"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
            role="alert"
            style="border-radius: 30px; justify-content: left; height: 220px; width: 480px; background-color: white; color: black; font-size: 18px; padding-left: 40px; padding-right: 40px; padding-top: 0px;"
          >
            <div
              class="MuiSnackbarContent-message"
            >
              message
            </div>
            <div
              class="MuiSnackbarContent-action"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                style="margin-right: 220px; margin-top: -40px; font-weight: 600; color: white; font-size: 15px;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  OK
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-backdrop-3"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
          role="alert"
          style="border-radius: 30px; justify-content: left; height: 220px; width: 480px; background-color: white; color: black; font-size: 18px; padding-left: 40px; padding-right: 40px; padding-top: 0px;"
        >
          <div
            class="MuiSnackbarContent-message"
          >
            message
          </div>
          <div
            class="MuiSnackbarContent-action"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              style="margin-right: 220px; margin-top: -40px; font-weight: 600; color: white; font-size: 15px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                OK
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
